main/Smartsheet/Api/FolderResources.cs

```diff
@@ -90,15 +92,14 @@ public interface FolderResources
 		/// <exception cref="ResourceNotFoundException"> if the resource cannot be found </exception>
 		/// <exception cref="ServiceUnavailableException"> if the REST API service is not available (possibly due To rate limiting) </exception>
 		/// <exception cref="SmartsheetException"> if there is any other error during the operation </exception>
-		IList<Folder> ListFolders(long parentFolderId);
+		DataWrapper ListFolders(long folderId, bool includeAll, long? pageSize, long? page);
```

Comment:
Instead of passing parameters of includeAll, pageSize, page instead create and pass in a  PaginationParameters object that holds that data. This is beneficial as the includeAll, pageSize and page are used in a lot of different resources so passing in only one object keeps things clean. Plus, we can then add a method to PaginationParameters which knows how to take the fields and convert them into a query string. This also means we will not need the `CreatePathAndQueryWithIncludeAndPaging()` and `CreatePathAndQueryWithPaging()` methods. Here is how we implemented this class in Java: https://github.com/smartsheet-platform/smartsheet-java-sdk/blob/release-2.0/src/main/java/com/smartsheet/api/models/PaginationParameters.java